unmarshal_test.go

```diff
@@ -0,0 +1,401 @@
+}
+
+func TestUnmarshal(t *testing.T) {
+	setEnvForTest(t, "STRING_FIELD", "string_value")
+	setEnvForTest(t, "INT_FIELD", "456")
+	setEnvForTest(t, "BOOL_FIELD", "false")
+	setEnvForTest(t, "FLOAT_FIELD", "4.56")
+	setEnvForTest(t, "SLICE_FIELD", "item1,item2")
+	setEnvForTest(t, "SLICE_FIELD_MULTI", "item1,item2,item3,item4")
+
+	var cfg Config
+	err := Unmarshal(&cfg)
+	assertNoError(t, err, "Unmarshal")
+
+	expected := Config{
+		StringField:              "string_value",
+		IntField:                 456,
+		BoolField:                false,
+		FloatField:               4.56,
+		StringSliceField:         []string{"item1", "item2"},
+		StringSliceFieldMultiple: []string{"item1", "item2", "item3", "item4"},
+	}
+
+	assertEqual(t, expected, cfg, "Unmarshal")
+}
+
+func TestUnmarshalWithDefaults(t *testing.T) {
+	var cfg Config
+	err := Unmarshal(&cfg)
+	assertNoError(t, err)
+	assertEqual(t, "default_value", cfg.StringField)
+	assertEqual(t, 123, cfg.IntField)
+	assertEqual(t, true, cfg.BoolField)
+	assertEqual(t, 1.23, cfg.FloatField)
+	assertEqual(t, []string{"item1"}, cfg.StringSliceField)
+	assertEqual(t, []string{"item1", "item2", "item3"}, cfg.StringSliceFieldMultiple)
+}
+
+func TestUnmarshalNested(t *testing.T) {
+	setEnvForTest(t, "STRING_FIELD", "string_value")
+	setEnvForTest(t, "INT_FIELD", "456")
+	setEnvForTest(t, "BOOL_FIELD", "false")
+	setEnvForTest(t, "FLOAT_FIELD", "4.56")
+	setEnvForTest(t, "SLICE_FIELD", "item1,item2")
+	setEnvForTest(t, "SLICE_FIELD_MULTI", "item1,item2,item3,item4")
+	setEnvForTest(t, "NESTED_NESTED_FIELD", "nested_value")
+
+	var cfg ParentConfig
+	err := Unmarshal(&cfg)
+	assertNoError(t, err, "Unmarshal")
+
+	expected := ParentConfig{
+		Config: Config{
+			StringField:              "string_value",
+			IntField:                 456,
+			BoolField:                false,
+			FloatField:               4.56,
+			StringSliceField:         []string{"item1", "item2"},
+			StringSliceFieldMultiple: []string{"item1", "item2", "item3", "item4"},
+		},
+		Nested: NestedConfig{
+			NestedField: "nested_value",
+		},
+	}
+
+	assertEqual(t, expected, cfg, "UnmarshalNested")
+}
+
+func TestUnmarshalUntaggedFields(t *testing.T) {
+	type UntaggedFieldsConfig struct {
+		UntaggedField1 string
+		UntaggedField2 int
+		TaggedField    string `env:"TAGGED_FIELD,default=default_value"`
+	}
+
+	setEnvForTest(t, "TAGGED_FIELD", "value")
+
+	var cfg UntaggedFieldsConfig
+	err := Unmarshal(&cfg)
+	assertNoError(t, err, "Unmarshal")
+
+	expected := UntaggedFieldsConfig{
+		UntaggedField1: "",
+		UntaggedField2: 0,
+		TaggedField:    "value",
+	}
+
+	assertEqual(t, expected, cfg, "UnmarshalUntaggedFields")
+}
+
+func TestUnmarshalFloat(t *testing.T) {
+	setEnvForTest(t, "FLOAT32_VALUE", "3.14")
+	setEnvForTest(t, "FLOAT64_VALUE", "3.14159")
+
+	var cfg struct {
+		Float32Value float32 `env:"FLOAT32_VALUE"`
+		Float64Value float64 `env:"FLOAT64_VALUE"`
+	}
+	err := Unmarshal(&cfg)
+	assertNoError(t, err, "Unmarshal FloatConfig")
```

Comment:
What about moving your asserter to a package

This way, you will use `assert.NoError`, `assert.Equal`…

It could be a local package, or a go module in a dedicated repository you would reuse in all your dev